search/lib/stac/extension.js

```diff
@@ -12,15 +12,12 @@ function stripStacExtensionsFromRequestObject (request) {
   return strippedRequestObject;
 }
 
-function applyStacExtensions (extensions, result) {
+function applyStacExtensions (result, options) {
   let resultToReturn = Object.assign({}, result);
-  if (_.hasIn(extensions, EXTENSION_TYPES.fields)) {
```

Comment:
Moved this logic into the fields extension, so we could in theory do something like 
```
[fieldExtension, contextExtension, …].reduce((resultToReturn, extension) => extension.apply(resultToReturn), Object.assign({}, result))
```